src/Sarif/Writers/ResultLogJsonWriter.cs

```diff
@@ -71,63 +74,25 @@ public void Initialize(Run run)
 
             _jsonWriter.WriteStartObject(); // Begin: run
 
-            if (run.AutomationDetails != null)
-            {
-                _jsonWriter.WritePropertyName("automationDetails");
-                _serializer.Serialize(_jsonWriter, run.AutomationDetails);
-            }
-
-            if (!string.IsNullOrEmpty(run.BaselineGuid))
-            {
-                _jsonWriter.WritePropertyName("baselineGuid");
-                _serializer.Serialize(_jsonWriter, run.BaselineGuid);
-            }
-
-            if (run.RunAggregates != null)
-            {
-                _jsonWriter.WritePropertyName("runAggregates");
-                _serializer.Serialize(_jsonWriter, run.RunAggregates);
-            }
-
-            if (run.Conversion != null)
-            {
-                _jsonWriter.WritePropertyName("conversion");
-                _serializer.Serialize(_jsonWriter, run.Conversion);
-            }
-
-            if (run.VersionControlProvenance != null)
-            {
-                _jsonWriter.WritePropertyName("versionControlProvenance");
-                _serializer.Serialize(_jsonWriter, run.VersionControlProvenance);
-            }
-
-            if (run.OriginalUriBaseIds != null)
-            {
-                _jsonWriter.WritePropertyName("originalUriBaseIds");
-                _serializer.Serialize(_jsonWriter, run.OriginalUriBaseIds);
-            }
-
-            if (run.DefaultEncoding != null)
-            {
-                _jsonWriter.WritePropertyName("defaultEncoding");
-                _serializer.Serialize(_jsonWriter, run.DefaultEncoding);
-            }
-
-            if (run.RedactionTokens != null)
-            {
-                _jsonWriter.WritePropertyName("redactionTokens");
-                _serializer.Serialize(_jsonWriter, run.RedactionTokens);
-            }
+            SerializeIfNotNull(_run.AutomationDetails, "automationDetails");
```

Comment:
>automationDetails [](start = 56, length = 17)

Where did your property ordering come from? May be preferable to keep it in sync with the schema since (as a result of our code gen ordering) this will drive log property ordering when using standard JSON.NET serialization.

Reply:
I tried to:
- Maintain the current split between InitializeRun and CompleteRun
- Continue to write the "special" components in the order the code was before.
- Write everything else alphabetically to make it easy to check for missing properties relative to the full Run list.

I can make it match the schema order, but only within the before and after results segments.

Reply:
Changed to move everything down to "CompleteRun" and to make everything ordered per schema except Results. 

This still won't be consistent with serializing from the whole OM because of the ordering of Results itself, and also will be inconsistent if the API user calls one of the ResultLogJsonWriter methods to write another collection early (ex: WriteArtifacts).